File: app/src/main/java/com/meiji/daily/module/postscontent/PostsContentActivity.java
package com.meiji.daily.module.postscontent;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;

import com.meiji.daily.R;
import com.meiji.daily.module.base.BaseNewActivity;

/**
 * Created by Meiji on 2017/12/6.
 */

public class PostsContentActivity extends BaseNewActivity {

    private static final String EXTRA_TITLEIMAGE = "EXTRA_TITLEIMAGE";
    private static final String EXTRA_TITLE = "EXTRA_TITLE";
    private static final String EXTRA_SLUG = "EXTRA_SLUG";

    public static void start(@NonNull Context context, @NonNull String titleImage, @NonNull String title, int slug) {
        Intent starter = new Intent(context, PostsContentActivity.class);
        starter.putExtra(EXTRA_TITLEIMAGE, titleImage);
        starter.putExtra(EXTRA_TITLE, title);
        starter.putExtra(EXTRA_SLUG, slug);
        context.startActivity(starter);
    }

    @Override
    protected int attachLayoutId() {
        return R.layout.container;
    }

    @Override
    protected void initViews() {

    }

    @Override
    protected void initData() {
        Intent intent = getIntent();
        if (intent == null) {
            finish();
            return;
        }
        String titleImage = intent.getStringExtra(EXTRA_TITLEIMAGE);
        String title = intent.getStringExtra(EXTRA_TITLE);
        int slug = intent.getIntExtra(EXTRA_SLUG, 0);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle(title);
        }
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.container, PostsContentNewView.newInstance(titleImage, title, slug))
                .commit();
    }
}

File: app/src/main/java/com/meiji/daily/module/postscontent/PostsContentActivity.kt
package com.meiji.daily.module.postscontent

import android.content.Context
import android.content.Intent
import android.os.Bundle

import com.meiji.daily.R
import com.meiji.daily.module.base.BaseActivity

/**
 * Created by Meiji on 2017/12/6.
 */

class PostsContentActivity : BaseActivity() {

    override fun attachLayoutId() = R.layout.container

    override fun initViews() {

    }

    override fun initData(savedInstanceState: Bundle?) {
        if (intent == null) {
            finish()
            return
        }
        val titleImage = intent.getStringExtra(EXTRA_TITLEIMAGE)
        val title = intent.getStringExtra(EXTRA_TITLE)
        val slug = intent.getStringExtra(EXTRA_SLUG)
//        if (supportActionBar != null) {
//            supportActionBar!!.title = title
//        }
        if (savedInstanceState == null) {
            supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.container, PostsContentView.newInstance(titleImage, title, slug))
                    .commit()
        }
    }

    companion object {

        internal val EXTRA_TITLEIMAGE = "EXTRA_TITLEIMAGE"
        internal val EXTRA_TITLE = "EXTRA_TITLE"
        internal val EXTRA_SLUG = "EXTRA_SLUG"

        fun start(context: Context, titleImage: String, title: String, slug: String) {
            val starter = Intent(context, PostsContentActivity::class.java)
            starter.putExtra(EXTRA_TITLEIMAGE, titleImage)
            starter.putExtra(EXTRA_TITLE, title)
            starter.putExtra(EXTRA_SLUG, slug)
            context.startActivity(starter)
        }
    }
}
